### PR TITLE
run pub for analyze and test

### LINKED ISSUE #2493: "flutter analyze" and "Flutter test" don't run "pub get"
I wrote a travis script that ran "flutter analyze", but it failed confusingly because pub get hadn't run.

### PR DESCRIPTION
**Pull request by devoncarew**
- run pub for `flutter analyze` and `flutter test`
- fix https://github.com/flutter/flutter/issues/2493

### CODE REVIEW

**Comment by devoncarew** [comment]
Oops, this needs some re-work.

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 1]
```diff
@@ -121,9 +121,13 @@ abstract class FlutterCommand extends Command {
     }
 
     if (_usesPubOption && argResults['pub']) {
-      int exitCode = await pubGet();
-      if (exitCode != 0)
-        return exitCode;
+      // Flutter analyze and flutter test can work both on a project and on the flutter repo.
+      bool flutterRepo = argResults.options.contains('flutter-repo') && argResults['flutter-repo'];
+      if (!flutterRepo) {
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart]: this is trickier than that. flutter analyze in particular can do both the current package _and_ the flutter repo.

Comment (devoncarew) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart]: Ah, good catch. I didn't realize it could do both at once.

**Comment by devoncarew** [comment]
ptal, updated to handle `analyze` better.

File: packages/flutter_tools/lib/src/commands/test.dart [hunk 2]
```diff
@@ -31,6 +32,9 @@ class TestCommand extends FlutterCommand {
   String get description => 'Run Flutter unit tests for the current project (Linux only).';
 
   @override
+  bool get shouldRunPub => argResults['flutter-repo'] ? false : super.shouldRunPub;
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/test.dart]: you could also write this as `!argResults['flutter-repo'] && super.shouldRunPub`.

**Comment by Hixie** [comment]
LGTM

File: packages/flutter_tools/lib/src/commands/test.dart (hunk 2 above)
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/test.dart]: done

**Comment by Hixie** [comment]
Thanks!

**Pull request merged**